### PR TITLE
♻ Refactor `logic` module into package

### PR DESCRIPTION
**Pull request by lig**
Refactor `logic` module into package in preparation for adding `issues` logic.

### CODE REVIEW

**Comment by lig** [comment]
please review :)

File: src/logic/base.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+from textwrap import indent
```
Comment (samuelcolvin) [inline src/logic/base.py:1]: surely this should all go in `__init__.py`?

**Review by samuelcolvin — COMMENTED** [review]
otherwise LGTM.

**Review by samuelcolvin — COMMENTED** [review]
otherwise LGTM.

File: src/logic/base.py (hunk 1 above)
Comment (lig) [reply src/logic/base.py:1]: If you mean everything in the module I just like to keep `__init__` clear from the actual code and just build package's api there. It's the same as with `__init__` and `main` in `pydantic`. I can call this module `common` or `main` if you'd prefer that.

Comment (samuelcolvin) [reply src/logic/base.py:1]: but this package is much smaller than pydantic, and it has a single entry point, not many. Hence I think all this logic should just go into `__init__.py`.

Comment (lig) [reply src/logic/base.py:1]: ok

Comment (lig) [reply src/logic/base.py]: Updated

Comment (lig) [reply src/logic/base.py]: Updated

Comment (lig) [reply src/logic/base.py]: Actually, I would argue to introduce `base` module back. As now `__init__` imports `prs` and will import `issues` and those are importing `models` and this introduces a circular import. Basically, I think this refactoring brings `logic` into the league of packages and we need to follow the rules 🤷🏻

**Review by samuelcolvin — APPROVED** [review]
LGTM

**Pull request merged**